src/operator/tensor/ordering_op-inl.h

```diff
@@ -401,8 +401,6 @@ void TopKImpl(const RunContext &ctx,
     mxnet::op::SortByKeyWorkspaceSize<int, int, xpu>(src.Size()));
   temp_size = std::max(temp_size,
     mxnet::op::SortByKeyWorkspaceSize<int, DType, xpu>(src.Size()));
-  temp_size = std::max(temp_size,
-    mxnet::op::SortByKeyWorkspaceSize<DType, int, xpu>(src.Size()));
```

Comment:
have you tried other dtypes like int64 ?

Reply:
topk supports only floating types for now

Reply:
yes, ran these tests successfully on my unix machine - https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_ndarray.py#L799 it runs as part of `test_ndarray.py:test_order`